pkg/rc/farm.go

```diff
@@ -210,8 +211,21 @@ START_LOOP:
 				// at this point the rc is ours, time to spin it up
 				rcLogger.NoFields().Infoln("Acquired lock on new replication controller, spawning")
 
+				// TODO: maybe we don't need to fetch the RC
+				// here, but that involves changing replication
+				// controller code which expands the scope of
+				// the change to watching RC keys rather than
+				// values.  Also it probably doesn't matter
+				// much since we won't actually be doing a
+				// fetch that often (only when lock not held)
+				rc, err := rcf.rcStore.Get(rcKey.ID)
```

Comment:
hmm yeah I hadn't thought about what happens when the server turns on. I do think it'll be okay though, I believe consul is very efficient at single key retrievals. The main issue if there was one would probably be the http server spending a lot of effort on tls setup but connection pooling would probably make that better? In any case I think it should be okay but probably this fetch could be removed if we fix up RC code